src/Microsoft.AspNet.Mvc.Core/ModelBinding/SimpleTypeModelBinder.cs

```diff
@@ -8,12 +8,10 @@
 
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
-    public sealed class TypeConverterModelBinder : IModelBinder
+    public class SimpleTypeModelBinder : IModelBinder
     {
         public async Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
-            ModelBindingHelper.ValidateBindingContext(bindingContext);
```

Comment:
We shouldn't have stuff like this in the code. If we care about this kind of validation, we'd choose a better approach.